app/models/account.rb

```diff
@@ -56,6 +57,11 @@ def level_set(step)
 
     save
   end
+
+  def hook_req
```

Comment:
Should try to put it into level_set

Reply:
@gfedorenko rewrite according to this changes